packages/runtime/src/useCases/consumption/attributes/CreateRepositoryAttribute.ts

```diff
@@ -39,6 +39,18 @@ export class CreateRepositoryAttributeUseCase extends UseCase<CreateRepositoryAt
                 ...request.content
             }
         });
+        const queryWithoutOptionalProperties = flattenObject({
```

Comment:
A more serious issue: If an attribute exists with an optional property set - and you try to create an attribute without that optional property (e. g. leaving out the state of an address), then it is blocked

Reply:
Hi I rename this to queryWithoutOptionalRequestProperties this only refers to optional properties of the Request : tags, validFrom, validTo for the attribute itself all combinations are still possible only direct duplicates are blocked

Reply:
Still not fond of that because it requires understanding that there are optional properties - also the query content is named in the code, there's no further need to explain what it is. What the query is for is more useful IMO.

Reply:
> A more serious issue: If an attribute exists with an optional property set - and you try to create an attribute without that optional property (e. g. leaving out the state of an address), then it is blocked

Should I push a test for that?

Reply:
yes!